### ISSUE TITLE
Is there best pratice or routine to tune the hyper-parameters?

### ISSUE THREAD

**Issue by MengHao666** [yifan123/flow_grpo#212]
doses anyone have expperience on adjusting hyper paramerters like Grpo-guard `Clip range` 、`LR`、`Noise level`、weight of kl loss ? If I use GRPO-GUARD, how to tune `config.train.clip_range` gradually to get best results?

I am now using flow-grpo to train qwen-image with full parameters. I found the RL experiments is very sensitive to hyper paramerters.
when i used `1e-4` as LR,  the result is as following. the model as going to hakcing in only 10 train steps. but when I lower the LR into `6e-5`, the model do not hack but learn very slowly.

common setting: `cps` as sde_type, `0.8` as Noise level  `Clip range`=1e-4 `Noise level` =0.8, use velocity mse loss as kl loss, the weight is 1e-5， without GRPO-GUARD，weight of kl loss= 1e-5

setting 1: `LR`=1e-4

<img width="2090" height="1648" alt="Image" src="https://github.com/user-attachments/assets/1ecec5c9-4d5b-44e2-bfdb-730215ea7db7" />

setting2:  `LR`=6e-5

<img width="2146" height="1638" alt="Image" src="https://github.com/user-attachments/assets/e2bb4baa-26dc-4ff2-848d-0fcf08531c6f" />

**Comment by yifan123** [comment]
Currently, I have only utilized the component from GRPO-GUARD shown below:

<img width="700" height="284" alt="Image" src="https://github.com/user-attachments/assets/fb167bd6-a3e6-48c5-a7ab-46554f33cc83" />

I did not use this part:

<img width="300" height="116" alt="Image" src="https://github.com/user-attachments/assets/45191238-76ad-44cc-92f5-a81bf7106a67" />

(Note: diving `dt` can make training unstable on some flagship models.)

Based on this setup, compared to FlowGRPO, you can increase the learning rate (`lr`). Here is a set of the latest hyperparameters I used with a closed-source model for your reference:

*   **Data Collection:** Collect 64 prompts per epoch.
*   **Rollout:** Generate 8 images per prompt (8 is sufficient in most cases; if training issues arise, you can increase this to 16).
*   **Training Batch:** Use 32 collected prompts per training step. This means each epoch involves training twice on different subsets of data.
*   **Hyperparameters:**
    *   `lr = 1e-5`
    *   `loss_clip_range = 5e-6`
    *   `cfg = 1`
    *   `kl_loss_weight = 5e-5`
*   **SDE Settings:**
    *   `sde_window = [0, 5]`
    *   Sample 3 consecutive steps within this window each time.
    *   Apply SDE only during these 3 steps.
    *   `noise_level = 0.8`

**Comment by MengHao666** [comment]
> Currently, I have only utilized the component from GRPO-GUARD shown below:
> 
> <img alt="Image" width="700" height="284" src="https://private-user-images.githubusercontent.com/37578514/532649358-fb167bd6-a3e6-48c5-a7ab-46554f33cc83.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc3NTgxMDMsIm5iZiI6MTc2Nzc1NzgwMywicGF0aCI6Ii8zNzU3ODUxNC81MzI2NDkzNTgtZmIxNjdiZDYtYTNlNi00OGM1LWE3YWItNDY1NTRmMzNjYzgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA3VDAzNTAwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiMWQyNDM0MDIzMWViZDRmMjE4ODc0MWZlMTlhODY5OWY5NWY3YzlkOWM3ZGVlOTMzMjY3MDljNjY2NzliZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RX33FkdfcgxjKpcO2qEwDpLZvzlXg2KZlORSWdUrUb0">
> I did not use this part:
> 
> <img alt="Image" width="300" height="116" src="https://private-user-images.githubusercontent.com/37578514/532649393-45191238-76ad-44cc-92f5-a81bf7106a67.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc3NTgxMDMsIm5iZiI6MTc2Nzc1NzgwMywicGF0aCI6Ii8zNzU3ODUxNC81MzI2NDkzOTMtNDUxOTEyMzgtNzZhZC00NGNjLTkyZjUtYTgxYmY3MTA2YTY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA3VDAzNTAwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmZjRhZmQxMTdkMzMyYjIyNGFiYzQ1ZTFlYmM3YzYxZDVhMWFiNmI0Y2Q5ZmU5OTMxMzIyOTYyNmQ3YjQ0MDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CZrf7SBeSehJKVVWZ3-hwHwgrYOMvGO0i1Ia-mdaaAk">
> (Note: diving `dt` can make training unstable on some flagship models.)
> 
> Based on this setup, compared to FlowGRPO, you can increase the learning rate (`lr`). Here is a set of the latest hyperparameters I used with a closed-source model for your reference:
> 
> * **Data Collection:** Collect 64 prompts per epoch.
> * **Rollout:** Generate 8 images per prompt (8 is sufficient in most cases; if training issues arise, you can increase this to 16).
> * **Training Batch:** Use 32 collected prompts per training step. This means each epoch involves training twice on different subsets of data.
> * **Hyperparameters:**
>   
>   * `lr = 1e-5`
>   * `loss_clip_range = 5e-6`
>   * `cfg = 1`
>   * `kl_loss_weight = 5e-5`
> * **SDE Settings:**
>   
>   * `sde_window = [0, 5]`
>   * Sample 3 consecutive steps within this window each time.
>   * Apply SDE only during these 3 steps.
>   * `noise_level = 0.8`

thanks, I will have a try!
